frontend/src/services/admin.service.ts

```diff
@@ -0,0 +1,73 @@
+import { ApiService } from "./api.service";
+import type { User } from "@/types/user";
+import type { UserApiFilters } from "@/types/admin";
+
+export interface PaginationParams {
+  page?: number;
+  limit?: number;
+}
+
+export interface PaginatedUsersResponse {
+  data: User[];
+  total: number;
+  page: number;
+  limit: number;
+  totalPages: number;
+}
+
+export class AdminService {
+  static async getUsers(
+    filters: UserApiFilters = {},
+    pagination: PaginationParams = {},
+  ): Promise<PaginatedUsersResponse> {
+    const queryParams = new URLSearchParams();
+
+    if (filters.includeDeleted) {
+      queryParams.append("includeDeleted", "true");
+    }
+    if (filters.includeBlocked) {
+      queryParams.append("includeBlocked", "true");
+    }
+    if (filters.onlyDeleted) {
+      queryParams.append("onlyDeleted", "true");
+    }
+    if (filters.onlyBlocked) {
+      queryParams.append("onlyBlocked", "true");
+    }
+    if (filters.search && filters.search.trim()) {
+      queryParams.append("search", filters.search.trim());
+    }
+    if (filters.sortBy) {
+      queryParams.append("sortBy", filters.sortBy);
+    }
+    if (filters.sortOrder) {
+      queryParams.append("sortOrder", filters.sortOrder);
+    }
```

Comment:
bahahah quest ce que tu nous a fait 

```suggestion
    const queryParams = new URLSearchParams(filters);
```

ca suffisait